src/UsgsAstroLsSensorModel.cpp

```diff
@@ -704,112 +666,76 @@ csm::ImageCoord UsgsAstroLsSensorModel::groundToImage(
-                  firstOffset, firstTime, lastOffset, lastTime, nextOffset, nextTime)
+     // Convert to detector line and sample
+     detectorLine = m_iTransL[0]
+                  + m_iTransL[1] * detectorView[0]
+                  + m_iTransL[2] * detectorView[1];
 
-      if (fabs(lastOffset - firstOffset) < pixelPrec) {
-        MESSAGE_LOG(m_logger, "groundToImage: Final firstOffset {}, lastOffset {}, pixelPre {}", firstOffset, lastOffset, pixelPrec)
-        break;
-      }
+     // Convert to image sample line
+     approxPt.line += detectorLine;
    }
 
-   // The computed pixel is the detector pixel, so we need to convert that to image lines
-   csm::ImageCoord calculatedPixel(detectorCoord.line + closestLine, detectorCoord.samp);
+   timei = getImageTime(approxPt);
+   detectorView = computeDetectorView(timei, groundPt, adj, desiredPrecision);
 
-   // Reintersect to ensure the image point actually views the ground point.
-   csm::EcefCoord calculatedPoint = imageToGround(calculatedPixel, height);
-   double dx = ground_pt.x - calculatedPoint.x;
-   double dy = ground_pt.y - calculatedPoint.y;
-   double dz = ground_pt.z - calculatedPoint.z;
-   double len = dx * dx + dy * dy + dz * dz;
+   // Invert distortion
+   double distortedFocalX, distortedFocalY;
+   applyDistortion(detectorView[0], detectorView[1], distortedFocalX, distortedFocalY,
+                   m_opticalDistCoeffs, m_distortionType, desiredPrecision);
+
+   // Convert to detector line and sample
+   detectorLine = m_iTransL[0]
+                + m_iTransL[1] * distortedFocalX
+                + m_iTransL[2] * distortedFocalY;
+   detectorSample = m_iTransS[0]
+                  + m_iTransS[1] * distortedFocalX
+                  + m_iTransS[2] * distortedFocalY;
 
-   if (achieved_precision) {
-      *achieved_precision = sqrt(len);
+   // Convert to image sample line
+   approxPt.line += detectorLine;
+   approxPt.samp = (detectorSample + m_detectorSampleOrigin - m_startingSample)
+                 / m_detectorSampleSumming;
+
+   if (achievedPrecision) {
+     *achievedPrecision = detectorLine;
    }
 
-   double preSquare = desired_precision * desired_precision;
-   if (warnings && (desired_precision > 0.0) && (preSquare < len)) {
-     MESSAGE_LOG(m_logger, "groundToImage: Desired precision not achieved {}", preSquare)
-     std::stringstream msg;
-     msg << "Desired precision not achieved. ";
-     msg << len << "  " << preSquare << "\n";
-     warnings->push_back(csm::Warning(csm::Warning::PRECISION_NOT_MET,
-                         msg.str().c_str(),
-                         "UsgsAstroLsSensorModel::groundToImage()"));
+   MESSAGE_LOG(m_logger, "computeViewingPixel: image line sample {} {}",
+                                approxPt.line, approxPt.samp)
+
+   if (warnings && (desiredPrecision > 0.0) && (achievedPrecision) && (*achievedPrecision > desiredPrecision))

```

Comment:
If the user doesn't pass in an achievedPrecisions, this should still happen. So, I would actually change it to check that `(abs(detectorLine) > desiredPrecision)`

Sorry about sending you on the wrong path with my last review :(